### ISSUE TITLE
the code implementation for Shuffle Unit is inconsistent with the paper

### ISSUE THREAD

**Issue by kkwalking** [megvii-model/ShuffleNet-Series#56]
I am a freshman in this field. I found that the code(blocks.py) implementation for Shuffle Unit is inconsistent with the paper.In the paer, the order of the unit is 1x1 GConv ,channel shuffle, 3x3 DWConv, 1x1Gconv. the implementation for that in the blocks.py:

 branch_main_1 = [
            # pw
            nn.Conv2d(inp, mid_channels, 1, 1, 0, groups=1 if first_group else group, bias=False),
            nn.BatchNorm2d(mid_channels),
            nn.ReLU(inplace=True),
            # dw
            nn.Conv2d(mid_channels, mid_channels, ksize, stride, pad, groups=mid_channels, bias=False),
            nn.BatchNorm2d(mid_channels),
        ]
        branch_main_2 = [
            # pw-linear
            nn.Conv2d(mid_channels, outputs, 1, 1, 0, groups=group, bias=False),
            nn.BatchNorm2d(outputs),
        ]

def forward(self, old_x):
        x = old_x
        x_proj = old_x
        x = self.branch_main_1(x)
        if self.group > 1:
            x = self.channel_shuffle(x)
        x = self.branch_main_2(x)
        if self.stride == 1:
            return F.relu(x + x_proj)
        elif self.stride == 2:
            return torch.cat((self.branch_proj(x_proj), F.relu(x)), 1)

**Comment by kkwalking** [comment]
I found the same issue and got the answer. Sorry.
